subsys/logging/Kconfig.backends

```diff
@@ -325,6 +325,41 @@ source "subsys/logging/Kconfig.template.log_format_config"
 
 endif # LOG_BACKEND_ADSP
 
+config LOG_BACKEND_CAVS_HDA
+	bool "cAVS HDA backend"
+	depends on SOC_FAMILY_INTEL_ADSP && DMA && DMA_CAVS_HDA
+	help
+	  Provide a logging backend which writes to a buffer and
+	  periodically flushes to hardware using ringbuffer like
+	  semantics provided by DMA_CAVS_HDA.
+
+if LOG_BACKEND_CAVS_HDA
+
+backend = CAVS_HDA
+backend-str = cavs_hda
+source "subsys/logging/Kconfig.template.log_format_config"
+
+config LOG_BACKEND_CAVS_HDA_SIZE
+	int "Size of ring buffer"
+	range 128 8192
+	default 4096
+	help
+	  Sets the ring buffer size cAVS HDA uses for logging. Effectively
+	  determines how many log messages may be written to in a period of time.
+	  The period of time is decided by how often to inform the hardware of
+	  writes to the buffer.
+
+config LOG_BACKEND_CAVS_HDA_FLUSH_TIME
+	int "Time in milliseconds to periodically flush writes to hardware"
+	range 1 10000
+	default 500
+	help
+	    The cAVS HDA backend periodically writes out its buffer contents
+	    to hardware by informing the DMA hardware the contents of the ring
+	    buffer.
```

Comment:
Not sure a timer is really the right thing here, that's overhead inside the device when really decisions about output flushing want to be part of the app.    If nothing else, you could make it configurable by e.g. treating "0" as meaning "disabled".  I think the watermark-based capacity flushing should be acceptable for basically all routine use, and if we "Need To Get Stuff Out Now" we're probably in some kind of error condition and want a facility to do a synchronous flush anyway.